### PR TITLE
Added block count polling function and updated createTopic page

### PR DESCRIPTION
**Pull request by chrisli30**

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -3,9 +3,9 @@ export default {
 };
 
 const siteConfig = {
-  siteName: 'ISOMORPHIC',
+  siteName: 'Bodhi Predication Market',
```
Comment (dwalintukan) [inline src/config.js]: Prediction

File: src/config.js [hunk 2]
```diff
@@ -3,9 +3,9 @@ export default {
...
   siteIcon: 'ion-flash',
-  footerText: 'Isomorphic ©2017 Created by RedQ, Inc',
+  footerText: 'Bodhi Predication Market ©2017 Created by Bodhi.Network',
```
Comment (dwalintukan) [inline src/config.js]: Prediction

File: src/containers/Topbar/Topbar.js [hunk 3]
```diff
@@ -85,12 +93,48 @@ class Topbar extends React.PureComponent {
...
+      setTimeout(startPoll, POOL_INTERVAL);
+    }());
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const {
+      onGetBlockCount, getBotBalance, selectedWalletAddress,
+    } = this.props;
+
+
```
Comment (dwalintukan) [inline src/containers/Topbar/Topbar.js]: 👾

File: src/containers/Topbar/Topbar.js [hunk 4]
```diff
@@ -16,6 +16,8 @@ const { Header } = Layout;
 const DROPDOWN_LIST_MAX_LENGTH = 8;
 const ADDRESS_TEXT_MAX_LENGTH = 11;
 const KEY_ADD_ADDRESS_BTN = 'add_address';
+const POOL_INTERVAL = 30000;
+const MINIMAL_GAS_FEE = 0.01;
```
Comment (dwalintukan) [inline src/containers/Topbar/Topbar.js]: This should be 0.1. default gas is `250000` * `0.0000004` (gasPrice) = 0.1 QTUM

File: src/containers/Topbar/Topbar.js [hunk 5]
```diff
@@ -85,12 +93,48 @@ class Topbar extends React.PureComponent {
...
+
+
+    // Call API to retrieve BOT balance if wallet addresses have changed
+    if (!_.isEqual(this.props.walletAddrs, nextProps.walletAddrs)) {
+      _.each(nextProps.walletAddrs, (addressObj) => {
+        const ownerAddress = addressObj.address;
+        let senderAddress;
+
+        // Determine an address that has enough qtum for gas to be a sender
+        if (addressObj.qtum >= MINIMAL_GAS_FEE) {
```
Comment (dwalintukan) [inline src/containers/Topbar/Topbar.js]: what about if a user wants to `callcontract` (no gas needed) from an address that has no qtum?

File: src/containers/Topbar/Topbar.js [hunk 6]
```diff
@@ -85,12 +93,48 @@ class Topbar extends React.PureComponent {
...
+          const objWithQtum = _.find(nextProps.walletAddrs, (item) => item.qtum >= MINIMAL_GAS_FEE);
+
+          if (objWithQtum) {
+            senderAddress = objWithQtum.address;
+          }
+        }
+
+        // Found any address with balance greater than minimal gas fee
+        if (senderAddress) {
+          getBotBalance(ownerAddress, senderAddress);
```
Comment (dwalintukan) [inline src/containers/Topbar/Topbar.js]: do we want to get the botBalance of every address? or just the one that is selected?

File: src/containers/oracle.js [hunk 7]
```diff
@@ -18,7 +18,7 @@ import topicActions from '../redux/topic/actions';
 const RadioGroup = Radio.Group;
 const QTUM = 'QTUM';
 const BOT = 'BOT';
-const DEFAULT_RADIO_VALUE = 0;
+const DEFAULT_RADIO_VALUE = 1;
```
Comment (dwalintukan) [inline src/containers/oracle.js]: curious why is this defaulting to 1 now?

File: src/containers/oracle.js [hunk 8]
```diff
@@ -158,70 +155,235 @@ class OraclePage extends React.Component {
...
+            messages: [
+              {
+                text: `Result setter ${oracle.resultSetterAddress}`,
+                type: 'default',
+              },
+              {
+                text: `Consensus Threshold ${oracle.consensusThreshold || 0}. This value indicates the amount of BOT needed for result setting.`,
+                type: 'default',
+              },
+
```
Comment (dwalintukan) [inline src/containers/oracle.js]: 👾

File: src/containers/oracle.js [hunk 9]
```diff
@@ -158,70 +155,235 @@ class OraclePage extends React.Component {
...
+              {
+                text: 'Upon clicking Confirm, you will need to wait for BOT token to get approved. Those BOT amount will automatically be used to set result afterwards.',
```
Comment (dwalintukan) [inline src/containers/oracle.js]: Should we also say `Please don't leave this screen`?

File: src/containers/oracle.js [hunk 10]
```diff
@@ -158,70 +155,235 @@ class OraclePage extends React.Component {
...
+              },
+              {
+                title: 'Result Setting',
+                description: `Block No. ${oracle.endBlock + 1} - ${oracle.resultSetEndBlock}`,
+              },
+              ],
+            },
+            messages: [
+              {
+                text: `Result setter ${oracle.resultSetterAddress}`,
```
Comment (dwalintukan) [inline src/containers/oracle.js]: suggestion: call the resultSetter the `Oracle` or `CentralizedOracle`?

File: src/containers/oracle.js [hunk 11]
```diff
@@ -158,70 +155,235 @@ class OraclePage extends React.Component {
...
+            type: 'warn',
+          });
+        }
+      } else if (status === 'VOTING' && token === BOT) {
+        const relatedOracles = _.filter(getOraclesSuccess, (item) => item.topicAddress === oracle.topicAddress);
+        const centralizedOracle = _.find(relatedOracles, (item) => item.token === QTUM);
+
+        config = {
+          name: 'VOTING',
+          breadcrumbLabel: 'Votting',
```
Comment (dwalintukan) [inline src/containers/oracle.js]: Voting

File: src/containers/oracle.js [hunk 12]
```diff
@@ -158,70 +155,235 @@ class OraclePage extends React.Component {
...
+                title: 'Topic created',
+                description: `Block No. ${(centralizedOracle && centralizedOracle.blockNum) || ''}`,
+              },
+              {
+                title: 'Betting',
+                description: `Block No. ${(centralizedOracle && centralizedOracle.blockNum + 1) || ''} - ${(centralizedOracle && centralizedOracle.endBlock) || ''}`,
+              },
+              {
+                title: 'Result Setting',
+                description: `Block No. ${(centralizedOracle && centralizedOracle.endBlock + 1) || ''} - ${(centralizedOracle && centralizedOracle.resultSetEndBlock)}`,
```
Comment (dwalintukan) [inline src/containers/oracle.js]: do we need to show the `Betting` and `Result Setting` info for a DOracle? Seems like extraneous information to me.

File: src/containers/oracle.js [hunk 13]
```diff
@@ -158,70 +155,235 @@ class OraclePage extends React.Component {
...
+                description: `Block No. ${oracle.blockNum} - ${oracle.endBlock}`,
+              },
+              ],
+            },
+            messages: [
+              {
+                text: `Consensus Threshold ${oracle.consensusThreshold || 0}. This value indicates the amount of BOT needed to fulfill current voting challenge.`,
+                type: 'default',
+              }, {
+                text: 'BOT tokens are needed for Voting. Upon clicking Confirm, you will need to wait for BOT token to get approved. Those amount will automatically be used to Vote afterwards.',
```
Comment (dwalintukan) [inline src/containers/oracle.js]: should we say `dont leave this screen`?

File: src/containers/oracle.js [hunk 14]
```diff
@@ -158,70 +155,235 @@ class OraclePage extends React.Component {
...
+        const relatedOracles = _.filter(getOraclesSuccess, (item) => {
+          console.log(item.topicAddress, oracle.topicAddress);
+          return item.topicAddress === oracle.topicAddress;
+        });
+        const centralizedOracle = _.find(relatedOracles, (item) => item.token === QTUM);
+        const decentralizedOracles = _.orderBy(_.filter(relatedOracles, (item) => item.token === BOT), ['blockNum'], ['asc']);
+
+        config = {
+          name: 'FINALIZING',
+          breadcrumbLabel: 'Voting',
```
Comment (dwalintukan) [inline src/containers/oracle.js]: Finalizing

File: src/redux/app/reducer.js [hunk 15]
```diff
@@ -4,38 +4,96 @@ import { getDefaultPath } from '../../helpers/urlSync';
...
+      let newState = state;
+      let combinedAddresses = [];
 
       if (action.value.result) {
         result = _.orderBy(_.map(action.value.result, (item) => ({
           address: item.address,
           qtum: item.amount,
         })), ['qtum'], ['desc']);
+
+
```
Comment (dwalintukan) [inline src/redux/app/reducer.js]: 👾

File: src/redux/app/saga.js [hunk 16]
```diff
@@ -39,9 +39,42 @@ export function* getBlockCountRequestHandler() {
...
+      const options = {
+        method: 'POST',
+        body: JSON.stringify({
+          owner,
+          senderAddress,
+        }),
+        headers: { 'Content-Type': 'application/json' },
+      };
+
+      const result = yield call(request, 'http://localhost:8080/balanceof', options);
```
Comment (dwalintukan) [inline src/redux/app/saga.js]: change to `/botbalance` once my API pr gets merged.

File: src/containers/Topbar/Topbar.js (hunk 6 above)
Comment (chrisli30) [reply src/containers/Topbar/Topbar.js]: We plan to show balance of every address in his wallet, that way he can see all balances in topbar dropdown.

File: src/containers/Topbar/Topbar.js (hunk 5 above)
Comment (chrisli30) [reply src/containers/Topbar/Topbar.js]: then he hits else case where we loop through his wallet addresses and find the first with enough qtum to pay for gas.

File: src/containers/oracle.js (hunk 7 above)
Comment (chrisli30) [reply src/containers/oracle.js]: Good call. If 1 the first option is selected after toggle. After testing I decide to change it back to 0 where no option is selected initially.

File: src/containers/oracle.js (hunk 9 above)
Comment (chrisli30) [reply src/containers/oracle.js]: Sure, I'm going to add that text to above button which could be more obvious.

Comment (chrisli30) [reply src/containers/oracle.js]: good point. Added above text

File: src/containers/oracle.js (hunk 12 above)
Comment (chrisli30) [reply src/containers/oracle.js]: For current version we don't separate betting and voting. All four phases are in the same flow so I think showing betting and result settings matches what shows on dashboard tab. Open for discussion.

File: src/containers/oracle.js (hunk 13 above)
Comment (chrisli30) [reply src/containers/oracle.js]: Added.

File: src/containers/oracle.js (hunk 14 above)
Comment (chrisli30) [reply src/containers/oracle.js]: No, finalizing is in the same phase(tab) as Voting. We don't show Finalizing on tabs/breadcrumb navigation.

File: src/components/bodhi-dls/cardVoting.js [hunk 17]
```diff
@@ -4,26 +4,57 @@ import { connect } from 'react-redux';
...
-
 class CardVoting extends Component {
   constructor(props) {
     super(props);
 
     this.state = {
       voteAmount: 0,
+      btnLoading: false,
+      btnText: '',
+      btnDisabled: false,
```
[the remark this answers is not in the document]
Comment (chrisli30) [reply src/components/bodhi-dls/cardVoting.js]: Created an enhancement issue for myself. Will address this after alpha release.

File: src/containers/Topbar/Topbar.js (hunk 3 above)
Comment (chrisli30) [reply src/containers/Topbar/Topbar.js]: I just realized what this crab means.. have been ignoring this icon for previous PRs...

File: src/components/bodhi-dls/cardVoting.js (hunk 17 above)
[the remark this answers is not in the document]
Comment (dwalintukan) [reply src/components/bodhi-dls/cardVoting.js]: 👍

File: src/containers/Topbar/Topbar.js (hunk 3 above)
Comment (dwalintukan) [reply src/containers/Topbar/Topbar.js]: haha its a space invader from the old arcade game

File: src/containers/Topbar/Topbar.js (hunk 5 above)
Comment (dwalintukan) [reply src/containers/Topbar/Topbar.js]: What I meant to say is something like this case:
a user had Qtum before and he bet/voted on something, but now he has 0 qtum. but he is on the completed TopicEvent page and would like to check his `withdraw` amount (which costs no gas/qtum) amount on the from this address.  this logic would not allow him to do any free `callcontract` calls.

**Comment by chrisli30** [comment]
All comments are addressed. Merging this one.

**Pull request merged**